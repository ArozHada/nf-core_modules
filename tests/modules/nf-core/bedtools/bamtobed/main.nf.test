nextflow_process {

    name "Test Process BEDTOOLS_BAMTOBED"
    script "modules/nf-core/bedtools/bamtobed/main.nf"
    process "BEDTOOLS_BAMTOBED"
    config "tests/modules/nf-core/bedtools/bamtobed/nextflow.config"
    tag "bedtools_bamtobed"

    test("Sarscov2 Illumina SE BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id:'test'], //meta map
                             file(params.test_data['sarscov2']['illumina']['test_single_end_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }
    }
}
