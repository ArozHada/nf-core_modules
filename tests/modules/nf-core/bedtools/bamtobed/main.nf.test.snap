{
    "Sarscov2 Illumina SE BAM": {
        "content": [
            [
                "test.bed:md5,a6a299bd39dc56225f8029c05ea97dcb",
                "versions.yml:md5,049e965c6f0ce4dc291bfbdd0191a1ca"
            ]
        ],
        "timestamp": "2023-04-11T19:31:05+0000"
    }
}